specification/attributes/discount_handling.md

```diff
@@ -0,0 +1,49 @@
+# Discount Handling
+
+A discount is a pricing construct where providers offer a reduced price for services. Providers may have many types of discounts, including but not limited to commercially negotiated discounts, commitment-based discounts when you agree to a certain amount of usage or spend, and bundled discounts where you receive free or discounted usage of one product or service based on the usage of another. Discount Handling is commonly used in scenarios like verifying discounts were applied and calculating cost savings.
+
+Some discount offers can be purchased from a provider to get reduced prices. The most common example is a commitment-based discount, where you "purchase" a commitment to use or spend a specific amount within a period. There are instances where a portion of the purchased discount may go unused, which results in lost savings and needs to be clearly identifiable at a granular level. To facilitate this, the purchase amount is amortized over the term the discount is applied to (e.g., 1 year).
+
+Amortization is a process used to break down and spread purchase costs over a period of time or term of use. When a purchase is applicable to resources, like commitment-based discounts, the amortized cost of a resource takes the initial payment and term into account and distributes it out based on the resource's usage, attributing the prorated cost for each unit of billing. Amortization enables users of billing data to distribute purchase charges to the appropriate audience in support of cost allocation efforts. Discount Handling for purchased commitments is commonly used for scenarios like calculating utilization and implementing chargeback for the purchase amount.
+
+While providers may use different terms to describe discounts, FOCUS identifies a discount as being a reduced price applied directly to a row. Any price or cost reductions that are awarded after the fact are identified as a "Credit" Charge Subcategory. One example might be when a provider offers a reduced rate after passing a certain threshold of usage or spend.
+
+All rows defined in FOCUS MUST follow the discount handling requirements listed below.
+
+## Attribute ID
+
+DiscountHandling
+
+## Attribute name
+
+Discount Handling
+
+## Description
+
+Indicates how to include and apply discounts to usage charges or rows.
+
+## Requirements
+
+* All applicable discounts SHOULD be applied to each row they pertain to and SHOULD NOT be negated in a separate row.
+* All discounts applied to a row MUST apply to the entire charge.
+  * If a discount only applies to a subset of the charge, then that portion of the charge MUST be split into a separate row. One with the discount and one without.
+  * If multiple discounts apply, there MUST be separate rows for each unique combination of applied discounts or undiscounted amount.
```

Comment:
I am not sure how this would work? Say their is a commitment based discount and a general percentage discount.....how is bullet number one and bullet number 4 adhered to?

Reply:
I believe:
- If the commitment-based discount and the general discount are combined together, they should apply to every row.
- If the general discount is applied after the fact based on some overall criteria like total cost exceeding a threshold, then the general discount should appear as a credit

Reply:
That's correct. Let me give a different example:
Let's say you have a 10% negotiated discount (A), a 10% bundled discount (B), and a $1/hour spend commitment purchased for $0.50/hour (C). For the sake of this example, let's say the negotiated and bundled discounts cap out and don't apply to all resources so I can show what happens with a partially undiscounted amount. All examples are $0.05/minute for hourly data...

| Scenario | Resource | Discounts | Quantity | ListCost | OnDemandCost | BilledCost | EffectiveCost |
|-----------|-------|------------|------|----|---------|------------|----------------|
| No discounts | Foo0 |  | 60 | $3.00 | $3.00 | $3.00 | $3.00 |
| Negotiated only | Foo1 | A | 60 | $3.00 | $2.70 | $2.70 | $2.70 |
| Commitment only | Foo2 | C | 20 | $1.00 | $1.00 | $0.00 | $0.50 |
| Uncommitted part | Foo2 |    | 40 | $2.00 | $2.00 | $2.00 | $2.00 |
| Commitment only | Foo3 | C | 10 | $0.50 | $0.50 | $0.00 | $0.25 |
| Unused commitment |     | C | 10 | $0.50 | $0.50 | $0.00 | $0,25 |
| Negotiated&nbsp;+&nbsp;committed | Foo4 | A+C | 22.222 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated&nbsp;+&nbsp;uncommitted | Foo4 | A | 37.778 | $1.89 | $1.70 | $1.70 | $1.70 |
| All discounts<sup>1</sup> | Foo5 | A+B+C | 24.691 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated&nbsp;+&nbsp;bundled | Foo5 | A+B | 5.309 | $0.24 | $2.15 | $2.15 | $2.15 |
| Negotiated only | Foo5 | A | 15 | $0.75 | $0.675 | $0.675 | $0.675 |
| All discounts capped out | Foo5 |  | 15 | $0.75 | $0.75 | $0.75 | $0.75 |

<sup>_* As we've defined SKUs, bundled discounts would be applied to the list price, so that's why it doesn't look like there's a discount applied compared to Foo4._</sup>

Reply:
That's great example to explain how commitment-related charges work.
If it's planned to have illustrative examples in FOCUS, it's a must-have there!

Reply:
I'll add this to supporting content for now. I like the idea of figuring out the best way to be more explicit with examples in the main spec. We should think about that after this release.